packages/runner/sync-customer-os-data/repository/interaction_event_repository.go

```diff
@@ -81,20 +79,20 @@ func (r *interactionEventRepository) MergeInteractionSession(ctx context.Context
 	return dbRecord.(*db.Record).Values[0].(string), nil
 }
 
-func (r *interactionEventRepository) MergeInteractionEvent(ctx context.Context, tenant string, syncDate time.Time, message entity.EmailMessageData) (string, error) {
+func (r *interactionEventRepository) MergeInteractionEvent(ctx context.Context, tenant, externalSystemId string, syncDate time.Time, message entity.EmailMessageData) (string, error) {
 	session := utils.NewNeo4jWriteSession(ctx, *r.driver)
 	defer session.Close(ctx)
 
-	query := "MERGE (ie:InteractionEvent_%s {externalId:$externalId, source:$source, channel:$channel}) " +
+	query := "MATCH (:Tenant {name:$tenant})<-[:EXTERNAL_SYSTEM_BELONGS_TO_TENANT]-(e:ExternalSystem {id:$externalSystemId}) " +
+		" MERGE (ie:InteractionEvent_%s {externalId:$externalId, source:$source, channel:$channel}) " +
+		" MERGE (ie)-[:IS_LINKED_WITH {externalId:$externalId, syncDate: $syncDate}]->(e)" +
```

Comment:
Should be:
MERGE (ie:InteractionEvent_%s)-[rel:IS_LINKED_WITH {externalId:$externalId}]->(e)
ON CREATE SET
rel.syncDate=$syncDate,
ie.channel=$channel,
ie.source=$source,
continue with existing set

double merge, not sure how it will impact "ON CREATE"

Reply:
externalId not needed in InteractionEvent itself

Reply:
i've changed the code to be a one line MERGE. the outcome is the same as 2 merges